sim-cli/src/main.rs

```diff
@@ -42,9 +48,12 @@ fn deserialize_f64_greater_than_zero(x: String) -> Result<f64, String> {
 #[derive(Parser)]
 #[command(version, about)]
 struct Cli {
+    /// Path to a directory containing simulation files, and where simulation results will be stored
+    #[clap(long, short, env = "SIM_LN_DATA_DIR", default_value = DEFAULT_DATA_DIR)]
+    data_dir: PathBuf,
     /// Path to the simulation file to be used by the simulator
-    #[clap(index = 1)]
-    sim_file: PathBuf,
+    #[clap(long, short, default_value = DEFAULT_SIM_FILE)]
+    sim_file: String,
```

Comment:
Why are we defining this as `String` while `data_dir` is `PathBuf`?

Reply:
so users just declare a file name and we always search for it within `data_dir`

Reply:
I mean this could be `PathBuf` and the code will still run fine, not sure why it got changed

Reply:
Keeping it as a `PathBuf` would allow a user to specify a different path in conflict with configured / default `data_dir`. Further, since the `sim_file` configuration is now a flag and not a required positional arg, we'd need `PathBuf` value parser if we are to have a default value for it.

Reply:
I may be missing your point, but I don't think that's actually the case.

Right now you can already specify `--sim-file <path>/sim.json`. Furthermore, changing it to `Pathbuf` does not require changing the default value (notice how `DEFAULT_DATA_DIR` is defined as a `&str` and `data_dir` is `PathBuf`.

Reply:
> Keeping it as a PathBuf would allow a user to specify a different path in conflict with configured / default data_dir

now I think this might be desirable. applied